src/constraint_system/composer.rs

```diff
@@ -27,6 +27,9 @@ use hashbrown::HashMap;
 use num_traits::{One, Zero};
 use std::collections::BTreeMap;
 
+#[cfg(feature = "trace")]
```

Comment:
Why do we want this feature? And also, in case is needed? Shouldn't we enable it at `Cargo.toml` level with a feature-flag like it's done for `parallel` and `asm`?

Reply:
This is there because those are only imported when `trace` is enabled. I think the `Cargo.toml` is a bit strange here with the `trace` flag and the `trace-print` flag. Usually, you want features to group dependencies or large sections of a library but in this case it seems to be that `trace` is used for only one function and doesn't relate to any dependencies. Maybe this function `check_circuit_satisfied` makes sense to return a boolean, but certainly, it is a little confusing. I hadn't seen this in my code review.

Reply:
Ohh that was my bad. I missread and thought BigNumber was added as dep. And therefore wanted to constrain it's usage via features in `Cargo.toml` directly.